lib/compeon/api_client/middlewares/jsonapi_request.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require_relative 'deep_hash_transformer'
+
+require 'json'
+
+module Compeon
+  class APIClient
+    module Middlewares
+      class JSONAPIRequestMiddleware < Faraday::Middleware
+        include DeepHashTransformer
+
+        Faraday::Request.register_middleware jsonapi_request: self
+
+        def call(env)
+          if env.body
+            body = deep_transform_keys(env.body) do |key|
```

Comment:
```suggestion
            body = DeepHashTransformer.deep_transform_keys(env.body) do |key|
```
Personal preference ahead.